Test for notifications backend

iamhks opened the pull request:
## Summary

Closes #688

## Changes

- Added required tests

iamhks committed:
need to add auth

iamhks committed:
ask for help

iamhks commented:
Hi @tuxology I have tried many things from my end, but unable to authorize my test case requests. Would be glad if you could review and suggest changes.
<img width="959" alt="Screenshot 2023-03-26 at 01 22 03" src="https://user-images.githubusercontent.com/43777817/227738584-99b86de5-cc74-48a4-9362-f0c5ca7983fe.png">

@kamthamc @NdibeRaymond @Ritsyy @khabdrick  Would be great if you all could spare some time as well.

chaitanya-cloudknox commented on zubhub_backend/zubhub/notifications/tests/test_views.py:
```diff
@@ -0,0 +1,126 @@
+        self.auth = {
+            "username": "test_dummy2",
+            "email": "test_dummy2@gmail.com",
+            "password1": "qwertyu12345",
```
Do not add username and passwords to the codebase. Get the username and passwords through environment variables

iamhks replied:
@chaitanya-cloudknox I am not adding credentials in the main project codebase but only in the test case files. Referring to the current test cases here: https://github.com/unstructuredstudio/zubhub/blob/master/zubhub_backend/zubhub/creators/tests/test_views.py

chaitanya-cloudknox commented on zubhub_backend/zubhub/notifications/tests/test_views.py:
```diff
@@ -0,0 +1,126 @@
+
+    def test_get_user_notifications(self):
+        rgst = self.client.post(reverse("creators:signup_creator"), self.auth, format="json")
+        res = self.client.get(reverse('notifications:user-notifications'))
```
Are there any steps required before completing registration (like email verification)

iamhks replied:
@chaitanya-cloudknox I just checked and once you sign-up a new user it directly logs you in without any necessity of email-verification

iamhks committed:
Merge branch 'master' of https://github.com/iamhks/zubhub into tests

iamhks committed:
added authentication

iamhks commented:
@tuxology Finally able to resolve the issue. I have updated the PR and here is the screenshot:

![image](https://user-images.githubusercontent.com/43777817/228041816-f986a3ad-721c-4722-ae3f-d097b664b39d.png)

Please review and for the PUT api the reason to use param like that is because it was failing with status code 415 (Referred here: https://stackoverflow.com/questions/15153048/django-rest-framework-http-put-failing-with-415-on-django-1-5 , the created Notifications Object was not working with json.dumps(data)) Hence, made a request through local and added the same json body.
